tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -0,0 +1,286 @@
+                assert(diff2 == -1);
+            }
+            STATIC_ASSERT(CountedCompare<Iter, ConstIter> == common_with<Iter, ConstIter>);
+            STATIC_ASSERT(CountedCompare<ConstIter, Iter> == common_with<Iter, ConstIter>);
+            if constexpr (common_with<Iter, ConstIter>) { // cross-type difference
+                counted_iterator<Iter> iter1{Iter{input + 1}, 2};
+                counted_iterator<ConstIter> iter2{ConstIter{input}, 3};
+
+                const same_as<iter_difference_t<Iter>> auto diff1 = iter1 - iter2;
+                assert(diff1 == 1);
+
+                const same_as<iter_difference_t<ConstIter>> auto diff2 = iter2 - iter1;
+                assert(diff2 == -1);
+            }
+            { // difference default sentinel
+                counted_iterator<Iter> iter1{Iter{input}, 2};
+
+                const same_as<iter_difference_t<Iter>> auto diff1 = iter1 - default_sentinel;
+                assert(diff1 == -2);
+
+                const same_as<iter_difference_t<Iter>> auto diff2 = default_sentinel - iter1;
+                assert(diff2 == 2);
+            }
+            if constexpr (random_access_iterator<Iter>) { // decrement assign
+                counted_iterator<Iter> iter{Iter{end(input)}, 0};
+                auto ref = iter;
+                assert(addressof(iter -= 3) == addressof(iter));
+                assert(iter.count() == 3);
+                assert(*iter == 3);
+                assert(iter.count() == ref.count() + 3);
+                assert(iter.base().peek() == prev(ref.base().peek(), 3));
+            }
+        }
+        // [counted.iter.cmp]
+        {
+            { // equality
+                counted_iterator<Iter> iter1{Iter{input}, 2};
+                counted_iterator<Iter> iter2{Iter{input}, 2};
+                counted_iterator<Iter> iter3{Iter{input + 1}, 1};
+                assert(iter1 == iter2);
+                assert(!(iter1 == iter3));
+            }
+            { // inequality
+                counted_iterator<Iter> iter1{Iter{input + 1}, 2};
+                counted_iterator<Iter> iter2{Iter{input}, 3};
+                counted_iterator<Iter> iter3{Iter{input}, 3};
+                assert(iter1 != iter2);
+                assert(!(iter2 != iter3));
+            }
+            { // equality default sentinel
+                counted_iterator<Iter> iter1{Iter{input}, 2};
+                counted_iterator<Iter> iter2{Iter{input}, 0};
+                assert(iter1 != default_sentinel);
+                assert(default_sentinel != iter1);
+                assert(iter2 == default_sentinel);
+                assert(default_sentinel == iter2);
+            }
+            { // spaceship
+                counted_iterator<Iter> iter1{Iter{input}, 2};
+                counted_iterator<Iter> iter2{Iter{input + 2}, 0};
+                const same_as<strong_ordering> auto result = iter1 <=> iter2;

```

Comment:
Include `<compare>` for `strong_ordering`?

Reply:
Every Standard Library header that defines a `<=>` operator is specified to transitively include `<compare>`, including (pun intended) `<iterator>`.

Reply:
Nothing changed